src/main/java/com/google/sps/servlets/softwareDialogue.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a random quote. */
+@WebServlet("/software-dialogue")
+public final class softwareDialogue extends HttpServlet {
+  private int currentIndex = 0;
+
+  private int getCurrentIndex(){
+    return this.currentIndex;
+  }
+
+  private void setCurrentIndex(int index){
```

Comment:
Is this setter/ getter method necessary? Since you are just using the index within the class, could probably be simplified to just accessing and incrementing this.currentIndex.